File: python/historical_export.py
from client import create_native_client
import compassiot.compass.v1.time_pb2 as time
import nativeconnect.api.v1.nativeconnect_pb2 as native



def main():
    client = create_native_client()

    request = native.HistoricalExportRequest(
        filter_vins=["LRW3F7FS9PC893577"],
        filter_type=native.HistoricalExportFilterType.TRIP,
        filter_dates=time.DateTimeRange(
            start=time.LocalDate(
                day=1,
                month=10,
                year=2023
            ),
            end=time.LocalDate(
                day=10,
                month=10,
                year=2023
            ),
            day_of_week=[
                time.DayOfWeek.MONDAY,
                time.DayOfWeek.WEDNESDAY,
                time.DayOfWeek.FRIDAY,
                time.DayOfWeek.SUNDAY
            ],
            hour_of_day=[i for i in range(24)]  # 0 to 23
        )
    )

    response = client.HistoricalExport(request)

    print(response)

     
if __name__ == "__main__":
    main()

File: typescript/src/historical_export.ts
import * as time from "@buf/compassiot_model.bufbuild_es/compassiot/compass/v1/time_pb"
import {
  HistoricalExportFilterType,
  HistoricalExportRequestSchema, HistoricalExportResponseSchema
} from "@buf/nativeconnect_api.bufbuild_es/nativeconnect/api/v1/nativeconnect_pb"

import {createNodeClient} from "./client"
import {create, toJsonString} from "@bufbuild/protobuf";

const client = createNodeClient()

const request = create(HistoricalExportRequestSchema, {
  filterVins: ["LRW3F7FS9PC893577"],
  filterDates: {
    start: {
      day: 1,
      month: 10,
      year: 2024
    },
    end: {
      day: 31,
      month: 10,
      year: 2024
    },
    dayOfWeek: [
      time.DayOfWeek.SATURDAY,
      time.DayOfWeek.SUNDAY
    ],
    hourOfDay: Array.from(Array(24).keys())  // 0 to 23
  },
  filterType: HistoricalExportFilterType.TRIP
})

const response = await client.historicalExport(request)
console.log(toJsonString(HistoricalExportResponseSchema, response, { prettySpaces: 2 }))
